pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -277,13 +276,53 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	}
 
 	cidrStrings := make([]string, 0)
-	var northInterfaces networkv1.NorthInterfacesAnnotation
-	var additionalNodeNetworks networkv1.MultiNetworkAnnotation
 
 	if len(instance.NetworkInterfaces) == 0 || (len(instance.NetworkInterfaces) == 1 && len(instance.NetworkInterfaces[0].AliasIpRanges) == 0) {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: Node %v has no ranges from which CIDRs can be allocated", node.Name)
 	}
+
+	defer func() {
+		if rerr != nil {
+			return
+		}
+		// update node's spec
+		if !reflect.DeepEqual(node.Spec, oldNode.Spec) {
+			for i := 0; i < cidrUpdateRetries; i++ {
+				if err = utilnode.PatchNodeCIDRs(ca.client, types.NodeName(node.Name), node.Spec.PodCIDRs); err == nil {
+					klog.InfoS("Set the node PodCIDRs", "nodeName", node.Name, "cidrStrings", node.Spec.PodCIDRs)
+					break
+				}
+			}
+			if err != nil {
+				nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRAssignmentFailed")
+				klog.ErrorS(err, "Failed to update the node PodCIDR after multiple attempts", "nodeName", node.Name, "cidrStrings", node.Spec.PodCIDRs)
+				rerr = multierror.Append(rerr, err)
+				return
+			}
+			err = utilnode.SetNodeCondition(ca.client, types.NodeName(node.Name), v1.NodeCondition{
+				Type:               v1.NodeNetworkUnavailable,
+				Status:             v1.ConditionFalse,
+				Reason:             "RouteCreated",
```

Comment:
naive questions, is this "RouteCreated" the condition that applies here?

Reply:
I am only refactoring the code, this event reason was always there, but from my understanding this is for backward compatibility.